File: src/_720/Solution.js
/**
 * @param {string[]} words
 * @return {string}
 */
var longestWord = function(words) {
    var ans = '', stack = [''], length = words.length, i, reg, wLength;

    words.sort();
    for(i = 0; i < length; i++) {
        wLength = words[i].length;
        if (wLength > stack.length) continue;
        while(wLength < stack.length) stack.pop();
        reg = new RegExp('^' + stack[stack.length - 1]);
        if (reg.test(words[i])) {
            stack.push(words[i]);
            if (wLength > ans.length) ans = words[i]; 
        }
    }

    return ans;
};

File: src/_720/Solution.java
package _720;

import java.util.Arrays;
import java.util.HashSet;

class Solution {
    public String longestWord(String[] words) {
        String ans = "";
        HashSet<String> set = new HashSet<>(words.length);
        Arrays.sort(words);
        for (String s : words) {
            if (s.length() == 1 || set.contains(s.substring(0, s.length()-1))) {
                if (ans.length() < s.length()) {
                    ans = s;
                }
                set.add(s);
            }
        }
        return ans;
    }
}